src/main/services/typeOfBusinessService.ts

```diff
@@ -30,4 +33,48 @@ export class TypeOfBusinessService extends GenericService {
             return Promise.reject(errorData);
         }
     }
+
+    async saveAcspData (session: Session, selectedOption: string): Promise<void> {
+        // eslint-disable-next-line camelcase
+        const email = session?.data?.signin_info?.user_profile?.email!;
+        // eslint-disable-next-line camelcase
+        const userId = session?.data?.signin_info?.user_profile?.id!;
+        let acspData: AcspData = session?.getExtraData(USER_DATA)!;
+        try {
+            if (acspData === undefined) {
+                acspData = {
+                    id: userId,
+                    typeOfBusiness: selectedOption,
+                    email: email
+                };
+
+                // save data to mongo for the first time
+                try {
+                    await postAcspRegistration(session, session.getExtraData(SUBMISSION_ID)!, acspData);
+                } catch (error: any) {
+                    logger.error("Error posting ACSP " + JSON.stringify(error));
+                    if (error.httpStatusCode === 409) {
+                        // retry with put request
+                        await putAcspRegistration(session, session.getExtraData(SUBMISSION_ID)!, acspData).catch(() => {
```

Comment:
I dont think we should retry with put. If the record already exist then we display the error that the operation was unsuccessful. We should not update the record which is already there.